tests/functional/test_install_config.py

```diff
@@ -40,6 +40,24 @@ def test_command_line_options_override_env_vars(script, virtualenv):
     assert "Getting page https://download.zope.org/ppix" in result.stdout
 
 
+def test_no_password_in_debug_message(script, virtualenv):
+    """
+    Test that password in the URL is not logged
+    """
+    password = "my_password"
+
+    script.environ['PIP_INDEX_URL'] = (
```

Comment:
Why the environ options instead of passing `'--index-url', 'https://user:{}@example.com/simple/'.format(password)` to script.pip ?

Reply:
@xavfernandez Can you confirm if this test hits the network as written? It would be a lot better IMO if it didn't as I think such cases should be reduced to a minimum.

@expobrain It might be better to write a unit test (so in the `unit` directory rather than `functional`) of `_get_html_response()` to avoid hitting the network. You can look at `test_get_legacy_build_wheel_path__no_names()` and nearby functions for examples of using the `caplog` fixture (since the code wouldn't be running in a separate subprocess).

Reply:
PS - it looks like there are already a number of unit tests of that function (search for `test_get_html_response_()`), and you can follow their example of mocking the session response.

Reply:
> Can you confirm if this test hits the network as written? It would be a lot better IMO if it didn't as I think such cases should be reduced to a minimum.

I confirm it should (I was actually thinking about suggesting to add the `network` marker to the test) and I don't see a way around that for a functional test.
An unit test could indeed also remove the need for network.

Reply:
FYI, here's an example of a test failing because of flakiness due to installing `INITools` in the test: https://dev.azure.com/pypa/pip/_build/results?buildId=5274&view=logs&jobId=c897e773-a04c-5d18-b461-2c6584d92c5a

Reply:
@cjerdonek I've moved the test into `unit`, please have a look at it